### ISSUE TITLE
IN operator in querybuilder causes an exception

### ISSUE THREAD

**Issue by samrocketman** [rbw0/pysnow#55]
```
Traceback (most recent call last):
  File "exporter.py", line 169, in <module>
    query = s.query(table='sc_catalog', query=QueryBuilder().field('sys_id').equals(catalogID))
  File "/home/sam/git/github/servicenow-servicecatalog-exporter/.venv/local/lib/python2.7/site-packages/pysnow/query.py", line 70, in equals
    return self._add_condition('IN', ",".join(value), types=[str])
  File "/home/sam/git/github/servicenow-servicecatalog-exporter/.venv/local/lib/python2.7/site-packages/pysnow/query.py", line 148, in _add_condition
    raise QueryTypeError("Invalid type passed to %s() , expected: %s" % (caller, types))
```

Here's the code which raised the exception:
https://github.com/samrocketman/servicenow-servicecatalog-exporter/blob/82f1981fc10e6d1a7d872864fb20a6321bca8073/exporter.py

**Comment by rbw0** [comment]
Hello!

Pass the `list` object directly to `equals()` instead:

```python
QueryBuilder().field('sys_id').equals(catalogID)
```

Also, make sure you're running the latest version (0.4.10).

/Robert

**Comment by samrocketman** [comment]
Hi @rbw0 I actually got a similar error passing a list so I tried a String and got the above error.  In both cases, QueryBuilder fails.

Bear in mind, my way of using `','.join(categoryID)` should work; shouldn't it?

**Comment by rbw0** [comment]
This works in both python version 2 and 3 running the latest version of pysnow:

```python
>>> import pysnow
>>> print(pysnow.__version__)
0.4.10
>>> from pysnow import QueryBuilder as qb
>>> print(qb().field('sys_id').equals(['id1', 'id2']))
sys_idINid1,id2
```

Verified that older versions fails with the exception you're having.
If you do `pysnow.__version__` I'm pretty sure you'll see a version lower than 0.4.10.

Just upgrade and it should work properly:
```bash
$ pip install pysnow --upgrade
```

I did find an actual bug when testing; you can't pass lists of integers to equals(). But this is not the issue you're having, right? I'll have this fixed in the next version anyway.

**Comment by samrocketman** [comment]
Here's the version I'm using when I get that error.  https://github.com/samrocketman/servicenow-servicecatalog-exporter/blob/master/requirements.txt#L7

**Comment by rbw0** [comment]
This is even tested for 2.6+ and 3.3+
https://github.com/rbw0/pysnow/blob/0d85cbcdcfa4825f2556f33ca00a6df53c5b1dc8/tests/test_query.py#L120

Could you `print(pysnow.__version__)` in the environment you're having this issue please?

**Comment by samrocketman** [comment]
```
$ python
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import pysnow
>>> print(pysnow.__version__)
0.4.10
>>> 
```
Let me try again because I started avoiding the QueryBuilder.  I'll try using it again and see if I can replicate the issue.

**Comment by samrocketman** [comment]
I figured out the problem.  Whatever type that is being returned by the API is unknown but apparently it's not a `str`.  So I type cast as I was appending.

Here's the result of the working patch -> https://github.com/samrocketman/servicenow-servicecatalog-exporter/commit/8ef1ee8ae4a678c95d8dc8d03c1ae0c6727ce5d6
